mypy/server/aststripnew.py

```diff
@@ -85,13 +85,16 @@ def prepare_implicit_var_patches(self, node: ClassDef) -> None:
         """Produce callbacks that re-add attributes defined on self."""
         for name, sym in node.info.names.items():
             if isinstance(sym.node, Var) and sym.implicit:
+                explicit_self_type = sym.node.explicit_self_type
+
                 def patch() -> None:
                     existing = node.info.get(name)
+                    current_existing = node.info.names.get(name)
                     # This needs to mimic the logic in SemanticAnalyzer.analyze_member_lvalue()
                     # regarding the existing variable in class body or in a superclass.
```

Comment:
Can you update to comment to also explain what is going on, rather than just pointing out to another method?

Reply:
OK, but I think it is important to have a note that these must be kept strictly in sync.